configs/projects/puppet-bolt.rb

```diff
@@ -46,11 +46,25 @@
   proj.directory proj.link_bindir
 
   if platform.is_fedora? && platform.os_version.to_i >= 28
-    # Disable shebang mangling for certain paths inside PDK.
+    # Disable shebang mangling for certain paths inside Bolt.
     # See https://fedoraproject.org/wiki/Packaging:Guidelines#Shebang_lines
     brp_mangle_shebangs_exclude_from = [
       ".*/opt/puppetlabs/bolt/private/ruby/.*",
-      ".*/opt/puppetlabs/bolt/share/cache/ruby/.*",
+      ".*/opt/puppetlabs/bolt/share/cache/ruby/.*"
+    ].join('|')
+
+    proj.package_override("# Disable shebang mangling of embedded Ruby stuff\n%global __brp_mangle_shebangs_exclude_from ^(#{brp_mangle_shebangs_exclude_from})$")
+
+    # Disable build-id generation since it's currently generating conflicts
+    # with system libgcc and libstdc++
+    proj.package_override("# Disable build-id generation to avoid conflicts\n%global _build_id_links none")
+  end
+
+  if platform.name =~ /^el-(8)-.*/
+    # Disable shebang mangling for certain paths inside Bolt.
```

Comment:
This appears to be building and working with the limited testing i've done. Can someone who understands what is actually happening with the "shebang mangling" help review? 

building with 
```
bundle exec build puppet-bolt el-8-x86_64
```
Testing on redhat 8 by scp'ing output and rpm installing
```
rpm -i puppet-bolt-1.14.0.8.g066673a-1.el8.x86_64.rpm
```

Reply:
I came up with those two paths to exclude by just trying to run, if it failed I would just exclude the path that seemed to cause the error. So basically I have no idea what i'm doing.

Reply:
VMpooler node with bolt installed is `so4r5vo5og9w2j2.delivery.puppetlabs.net` package built with 
https://github.com/puppetlabs/bolt-vanagon/pull/86/commits/2b3cadad6653efb87d586a2acac9d8098ac4f52f
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt-vanagon-1/output/el/8/products/x86_64$ ls
puppet-bolt-1.14.0.8.g066673a-1.el8.x86_64.rpm
puppet-bolt-1.14.0.9.g2b3cada-1.el8.x86_64.rpm
cas@cas-ThinkPad-T460p:~/working_dir/bolt-vanagon-1/output/el/8/products/x86_64$ ssh so4r5vo5og9w2j2.delivery.puppetlabs.net
Last login: Wed Dec 12 10:20:57 2018 from 192.168.27.1
[root@so4r5vo5og9w2j2 ~]# ls
anaconda-ks.cfg  puppet-bolt-1.14.0.9.g2b3cada-1.el8.x86_64.rpm
original-ks.cfg
[root@so4r5vo5og9w2j2 ~]# rpm -i puppet-bolt-1.14.0.9.g2b3cada-1.el8.x86_64.rpm
```

Reply:
I claim ignorance around shebang mangling.

Reply:
Seems that from https://access.redhat.com/documentation/en-US/Red_Hat_Software_Collections/2/html-single/Packaging_Guide/#sect-Software_Collection_Shebang_Support the intent is to replace `/usr/bin/env thing` with `/absolute/path/to/thing` I'm wondering if maybe the failures are because there is no `/absolute/path/to/thing` indicating there are unsatisfied dependencies?

So specifically for the `/opt/puppetlabs/bolt/ssl/` exclusion I recall the error appeared to be something about `perl`. But perl appears to be on path of a redhat 8 instance. 
```
[root@so4r5vo5og9w2j2 ~]# which perl
/usr/bin/perl
``` 
So in that case i'm wondering if it is just safe to assume that given the package built and appears to work that `usr/bin/env perl` used (or i guess will use?) `/usr/bin/perl`.

Reply:
Ah ha. That makes sense.

/usr/bin/perl, IMO, is a very safe bet on any recent linux distro assuming no funky required modules.  I'll have another look

Reply:
After further review, it seems that if the package builds and installs, any internal shebang mangling is not an issue.

Reply:
OK, I will clean up this commit and request review on the puppet-agent PR.

So in terms of next steps we have the following PRs to merge
1. https://github.com/puppetlabs/puppet-runtime/pull/147
2. https://github.com/puppetlabs/ci-job-configs/pull/5672
3. This PR
I think that we have to merge 1 first. I think doing that will trigger a run of bolt-runtime CI that *wont* pick up the changes from 2 (which i think is good). It is unclear to me if merging 2 will trigger a run of bolt-vanagon CI (if it does there will be failures because the work in this PR wont be available). Finally once 1 and 2 are merged I think we could merge this one which i'm pretty sure would run with the necessary changes to puppet-runtime and CJC.

Reply:
Maybe I'm mistaken but I was under the impression that *this* PR should go in before the c-j-c config one.

Reply:
So I needed to run CJC experimental pipeline ( https://jenkins-master-prod-1.delivery.puppetlabs.net/view/__experimental%20automatic/job/experimental_auto_bolt-runtime_runtime-vanagon-init_master/2/ ) with custom parameters pointing to my fork of puppet-runtime to generate http://builds.delivery.puppetlabs.net/puppet-runtime/ef250d05fafc2a582fc4298d4826ea6e231ea954/artifacts/ which are used when we build bolt here.

Reply:
I think this does need to go in before the cjc change, since without this, vanagon won't know how to build for el 8.

Reply:
Does a merge to CJC master trigger all affected CI runs? I was assuming no. I was thinking that the change to CJC that depends on this would have its *first* run triggered by merging this work in (which will have the definitions for redhat 8). 

Also I think getting the order slightly wrong is low risk, worst case is that there is a round of failed jobs for redhat 8.

Reply:
I don't think cjc merge should kick off any jobs, if that's what you're asking. I was just imagining a scenario where cjc is updated and the job runs for another reason (other changes in this repo, someone kicks it, whatever your processes are).
But yeah, ultimately probably not a big deal either way. These should go in more-or-less at the same time, I think.

Reply:
Actually I think @mwaggett is correct. I think this needs to be merged first, then CJC, then puppet-runtime. So from the ordering above, 3,2,1.